src/main/java/fr/cnes/sonarqube/plugins/framac/sensor/FramaCSensor.java

```diff
@@ -61,6 +63,26 @@ public void describe(final SensorDescriptor sensorDescriptor) {
         sensorDescriptor.createIssuesForRuleRepositories(FramaCRulesDefinition.getRepositoryKeyForLanguage());
     }
 
+    /**
+     * Check if a file is a CSV file by analysing the header (first line)
+     * @param filePath : the file path
+     * @return true if the file is a csv file, false otherwise
+     */
+    private boolean isCsvFile(String filePath) {
+        boolean isCsvFile = false;
+        try (BufferedReader br =
+            new BufferedReader(new FileReader(filePath))) {
+            String line = br.readLine();
+            if (line.startsWith("directory\tfile")) {
+                isCsvFile = true;
+            }
+        } catch (IOException e) {
+            // Ignore (error will be processed later
+            isCsvFile = false;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this useless assignment; "isCsvFile" already holds the assigned value along all execution paths. ([squid:S4165](https://sonarcloud.io/organizations/lequal/rules#rule_key=squid%3AS4165))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fr.cnes.sonarqube.plugins%3Asonar-frama-c-plugin&pullRequest=23&issues=AWY0I9xNvOWp7Yd6e3Vh&open=AWY0I9xNvOWp7Yd6e3Vh)

[ISSUE_KEY]: AWY0I9xNvOWp7Yd6e3Vh